app/assets/javascripts/components/uploads/upload_list.jsx

```diff
@@ -62,31 +65,34 @@ const UploadList = createReactClass({
     let uploadsView;
 
     if (this.props.view === GALLERY_VIEW) {
-      uploadsView = (
-        <div className="gallery-view">
-          {elements}
-        </div>
-      );
+      if (elements.length === 0) {
+        uploadsView = (<div className="list-view">{noUploadsMessage}</div>);
+      } else {
+        uploadsView = (<div className="gallery-view"> {elements} </div>);
+      }
     }
 
     if (this.props.view === LIST_VIEW) {
-      uploadsView = (
-        <div className="list-view">
-          <List
-            elements={elements}
-            keys={keys}
-            table_key="uploads"
-            sortBy={this.props.sortBy}
-            none_message={I18n.t('courses_generic.uploads_none')}
-          />
-        </div>
-      );
+      if (elements.length && elements.length > 0) {
+        uploadsView = (
+          <div className="list-view">
+            <List
+              elements={elements}
+              keys={keys}
+              table_key="uploads"
+              sortBy={this.props.sortBy}
+            />
+          </div>);
+      } else {
+          uploadsView = (<div className="list-view">{noUploadsMessage}</div>);
+        }
     }
 
     if (this.props.view === TILE_VIEW) {
       uploadsView = (
         <div className="tile-view">
-          {elements}
+          {uploads.length > 0 && elements}
```

Comment:
Do you prefer doing it this way over combining this to one statement so that it's clearer that it will show `elements` OR `noUploadsMessage`? For example, `uploads.length > 0 ? elements : noUploadsMessage`.